storage/backend/ceph/ceph.go

```diff
@@ -321,6 +320,32 @@ func (c *Driver) ListSnapshots(do storage.DriverOptions) ([]string, error) {
 	return names, nil
 }
 
+// CopySnapshot copies a snapshot into a new volume. Takes a DriverOptions,
+// snap and volume name (string). Returns error on failure.
+func (c *Driver) CopySnapshot(do storage.DriverOptions, snapName, newName string) error {
+	cmd := exec.Command("rbd", "snap", "protect", do.Volume.Name, "--snap", snapName, "--pool", do.Volume.Params["pool"])
+	er, err := executor.NewWithTimeout(cmd, do.Timeout).Run()
+	if err != nil {
+		return err
+	}
+
+	if er.ExitStatus != 0 {
+		return fmt.Errorf("Protecting snapshot for clone (volume %q, snapshot %q): %v", do.Volume.Name, snapName, err)
+	}
+
+	cmd = exec.Command("rbd", "clone", do.Volume.Name, newName, "--snap", snapName, "--pool", do.Volume.Params["pool"])
+	er, err = executor.NewWithTimeout(cmd, do.Timeout).Run()
+	if err != nil {
```

Comment:
I think we should unprotect the sanpshot if clone fails, right?

On a related note, does the snapshot needs to stay protected after clone is successful?